src/toil/batchSystems/gridengine.py

```diff
@@ -299,7 +299,7 @@ def getRunningBatchJobIDs(self):
         return self.worker.getRunningJobIDs()
 
     def getUpdatedBatchJob(self, maxWait):
-        i = self.updatedJobsQueue.get()
+        i = self.getFromQueueSafely(self.updatedJobsQueue, maxWait)
```

Comment:
AFAIK, `getFromQueueSafely` was only necessary for `multiprocessing.Queue`, but not `Queue.Queue`. There is no need for these mostly IO bound threads in the batch systems to be multiprocessing processes. The the source to getFromQueueSafely does not specify the exact issue it is trying to work around. I am inclined to remove all mention of multiprocessing. GridEngine and Parasol are so slow that a little bit of GIL contention isn't going to make a difference.